contentcuration/contentcuration/viewsets/channel.py

```diff
@@ -321,7 +325,15 @@ def update(self, instance, validated_data):
                 instance.bookmarked_by.add(user_id)
             elif bookmark is not None:
                 instance.bookmarked_by.remove(user_id)
-        return super(ChannelSerializer, self).update(instance, validated_data)
+        was_deleted = instance.deleted
+        instance = super(ChannelSerializer, self).update(instance, validated_data)
+        # mark the instance as deleted or recovered, if requested
+        if user_id is not None and is_deleted is not None and is_deleted != was_deleted:
+            if is_deleted:
+                instance.mark_deleted(user_id)
```

Comment:
I'll remember to update this in my PR as well after merging - where I made the regular delete a soft delete too.